services/user/CommonApi/src/CommonApi.cpp

```diff
@@ -79,6 +84,23 @@ namespace SystemService
                 .body        = psio::convert_to_frac(trx),
             };
          }
+         if (request.target == "/common/set-cookie")
+         {
+            request.body.push_back(0);
+            psio::json_token_stream jstream{request.body.data()};
+            auto                    params = psio::from_json<cookie_data>(jstream);
+
+            // Create response headers with host-only __Host-SESSION cookie
+            std::vector<HttpHeader> headers;
+            std::string             cookieValue = "__Host-SESSION=" + params.accessToken +
+                                      "; Path=/; HttpOnly; Secure; SameSite=Strict";
+            headers.push_back({"Set-Cookie", cookieValue});
+
+            return HttpReply{.status      = HttpStatus::ok,
```

Comment:
This is missing the check for the origin to determine if it's a development chain, in which case the cookie should be such that it can be set without configuring the browser in a special way.

Reply:
The cookie names should probably be consistent, e.g.:
`__HOST-SESSION` and `SESSION`
or 
`__HOST-authed-query` and `authed-query`